### DISCUSSION TITLE
Error while configuring CMake

### DISCUSSION THREAD

**Discussion by Aques** [bastianvandenbout/LISEM#6] [Q&A] [answered]
Hello I used [Lisem Compile Yourself](https://lisemmodel.com/docs/compiling/) to compile Lisem myself. I downloaded mysys2 and installed all required packages using mysys2 CLI.
I Installed all the multi-part external dependencies zip and extracted it and placed it into external folder.
I also installed MPI and installed it using the Microsoft Installer.
In Step 5 for configuring CMake I gave following for Source and Destination,
Source: D:\Project\Lisem-main  (Lisem-main is the folder in which all files of Main branch are placed)
Destination: D:\Project\Lisem-bin
I also gave external folder path as needed.
But I am getting problem while configuring the CMake.
It gives error of unable to find MPI.
Kindly help

**Comment by bastianvandenbout** [comment, marked as answer]
Glad to hear you find some use with LISEM.
I see, I think I encountered this several times as well, although I am not so sure about a reliable fix.
Some things come to mind: It might be good to set the msys2/mingw64/bin directory in your environment PATH variable.
Restart the pc/windows/cmake etc.. to make sure cmake can find the neccesary files.
Confirm that the MSYS2 mpi package is installed as well as the Microsoft MPI installer.
It might help to manually set some variables in cmake to point it towards cmake:

![Untitled](https://user-images.githubusercontent.com/1472442/192975100-24b01ef7-58da-4994-99c8-341db5f5d118.png)

Finally, MPI and fortran (there is a small section of fortran in the LISEM source) use required some additional step in the past 
https://www.scivision.dev/windows-mpi-msys2/

So it might make sense to add microsoft MPI also to your environment PATH variable

Hope this helps, let me know!

**Reply by Aques** [reply]
Thank you so much for such a fast and detailed reply,
The solution given in reply worked.

- The problem was with CMAKE making wrong entry for msmpi library path. 
- So I deleted the entry and created MPI_msmpi_LIBRARY and set it to the path of libmsmpi.dll.a
